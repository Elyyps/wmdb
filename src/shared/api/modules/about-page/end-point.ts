import { generateAboutPageDummyData } from "./generate-dummy-data";

export const generateAboutPageData = generateAboutPageDummyData;
