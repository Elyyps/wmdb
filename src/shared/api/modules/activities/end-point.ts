import { generateActivitiesDummyData } from "./dummy-data";

export const generateActivitiesData = generateActivitiesDummyData;
