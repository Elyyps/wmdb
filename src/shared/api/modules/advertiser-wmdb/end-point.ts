import { generateAdvertiserDummyData } from "./dummy-data";
export const generateAdvertiserData = generateAdvertiserDummyData;
