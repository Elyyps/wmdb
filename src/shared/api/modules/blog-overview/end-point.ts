import { generateBlogOverviewDummyData, generateCards } from "./dummy-data";
import { ICompanyOutingCard } from "@app/api/core/company-outing-card";

export const generateBlogOverviewData = generateBlogOverviewDummyData;
const amountOfCards = 50;
export const getCardsWithPagination = (skip: number, take: number): ICompanyOutingCard[] => {
  let cards: ICompanyOutingCard[] = generateCards(amountOfCards);
  cards = cards.slice(skip, take + skip);

  return cards;
};
