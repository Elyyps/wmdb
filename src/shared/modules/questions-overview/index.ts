export * from "./questions-overview.component";
