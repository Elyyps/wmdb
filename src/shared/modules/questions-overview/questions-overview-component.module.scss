.questions-overview {
  &-body {
    background-color: #e8f2fc;
  }
}
