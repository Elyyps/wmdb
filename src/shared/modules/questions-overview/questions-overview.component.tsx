import * as React from "react";
import styles from "./questions-overview-component.module.scss";
import { IQuestionsOverview } from "@app/api/modules/questions-overview/questions-overview";
import { Breadcrumb } from "@app/core/breadcrumb";
import { ArrowPanelListComponent } from "@app/core/arrow-panel-list";
import { ModuleSectionComponent } from "@app/core/module-section";

export interface IQuestionsOverviewComponentProps {
  questionOverview: IQuestionsOverview;
}

const QuestionsOverviewComponent = ({ questionOverview }: IQuestionsOverviewComponentProps) => (
  <div className={styles["questions-overview"]}>
    <div className="uk-container">
      <Breadcrumb breadcrumbs={questionOverview.path} />
      <div className={styles["questions-overview-header"]}>
        <h1>{questionOverview.title}</h1>
        <p>{questionOverview.content}</p>
      </div>
    </div>
    <div className={styles["questions-overview-body"]}>
      <ModuleSectionComponent paddingTop="40px">
        <ArrowPanelListComponent items={questionOverview.link} />
      </ModuleSectionComponent>
    </div>
  </div>
);

export { QuestionsOverviewComponent };
