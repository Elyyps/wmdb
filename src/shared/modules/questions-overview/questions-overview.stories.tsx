import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { QuestionsOverviewComponent } from "./questions-overview.component";
import { generateQuestionsOverviewData } from "@app/api/modules/questions-overview/end-point";

storiesOf("QuestionsOverview", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <QuestionsOverviewComponent questionOverview={generateQuestionsOverviewData()} />);
