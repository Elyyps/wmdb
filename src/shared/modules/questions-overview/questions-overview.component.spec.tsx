import React from "react";
import { shallow } from "enzyme";
import { QuestionsOverviewComponent } from "./questions-overview.component";
import { generateQuestionsOverviewData } from "@app/api/modules/questions-overview/end-point";

describe("[QuestionsOverview]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(
      <QuestionsOverviewComponent questionOverview={generateQuestionsOverviewData()} />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
