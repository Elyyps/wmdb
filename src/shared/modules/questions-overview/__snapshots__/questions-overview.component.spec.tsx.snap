// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[QuestionsOverview] should render component without crashing 1`] = `
<div
  className="questions-overview"
>
  <div
    className="uk-container"
  >
    <Breadcrumb
      breadcrumbs={
        Array [
          Object {
            "title": "Home",
            "url": "/",
          },
          Object {
            "title": "FAQ",
            "url": "/",
          },
          Object {
            "title": "Feest",
            "url": "/",
          },
        ]
      }
    />
    <div
      className="questions-overview-header"
    >
      <h1>
        Alle veelgestelde vragen over feest
      </h1>
      <p>
        In onze FAQ belichten we alle facetten bij het organiseren van je evenement. Antwoorden op vragen en veel handige toepasbare tips.
      </p>
    </div>
  </div>
  <div
    className="questions-overview-body"
  >
    <ModuleSectionComponent
      paddingTop="40px"
    >
      <ArrowPanelListComponent
        items={
          Array [
            Object {
              "title": "Is er een checklist voor het plannen van een evenement?",
              "url": "",
            },
            Object {
              "title": "Wanneer moet u een evenementenverzekering afsluiten?",
              "url": "",
            },
            Object {
              "title": "Is een bedrijfsuitje anders dan een teamuitje?",
              "url": "",
            },
            Object {
              "title": "Wat zijn de belangrijkste tips bij het organiseren van een event?",
              "url": "",
            },
            Object {
              "title": "Hoe organiseert u een goed zakelijk jubileum?",
              "url": "",
            },
          ]
        }
      />
    </ModuleSectionComponent>
  </div>
</div>
`;
